File: cp_nim/src/bin/abc448-b-pepper-addiction.nim
# Created by Ayush Biswas at 2026/07/18 19:00
# https://atcoder.jp/contests/adt_easy_20260407_2/tasks/abc448_b
import ../lib/cpio

# @code begin

import math

var r = newReader()

var
  (n, m) = r[int, int]
  limits = r[seq[int]]
  peppers = r[seq[seq[int]], n]

var res: int
for pepper in peppers:
  res += limits[pepper[0] - 1].min pepper[1]
  limits[pepper[0] - 1] = max(limits[pepper[0] - 1] - pepper[1], 0)

echo res

# @code end

File: cp_rust/src/bin/abc448-b-pepper-addiction.rs
// Created by Ayush Biswas at 2026/07/07 21:35
// https://atcoder.jp/contests/adt_easy_20260407_2/tasks/abc448_b
use cp_lib::*;

// @code begin
use cpio::*;

sol! {
    fn solution(
        [n, m]: [usize; 2],
        (mut peppers): [usize],
        abs: [[usize; 2]; n]
    ) -> usize {
        let mut res = 0;
        for [a, b] in abs {
            let sprinkl = peppers[a - 1].min(b);
            res += sprinkl;
            peppers[a - 1] -= sprinkl;
        }
        res
    }
}
// @code end
